### ISSUE TITLE
[Workflow] Symfony Workflow unknown twig methods 

### ISSUE THREAD

**Issue by wnoveno** [symfony/symfony#26068]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.4.4
| Twig Extensions version  | 1.4

```
 {% if workflow_can(object, 'Draft') %}
    <a href="#">Reject article</a>
 {% endif %}
```
Error Message:
Did you forget to run "composer require symfony/workflow"? Unknown function "workflow_transitions".

Workflow is in composer and working on the Controller Side. 
But throws an error when using the twig methods.

*Edit: Tested other twig methods (workflow_can) as well and on other versions Symfony3.2. Returns the same error

**Comment by xabbuh** [comment]
Are you able to create a small application that allows to reproduce your issue?

**Comment by lyrixx** [comment]
hm, the error message is not perfect. 
you should install https://github.com/symfony/twig-bridge
Could you try and tell us if it's OK ?

BTW, how did you set-up your project With Symfony 3.2 ? because before SF 3.4, usually people install "everything": symfony/symfony. Here it looks like your are installing packages per packages.

**Comment by xabbuh** [comment]
> hm, the error message is not perfect.
> you should install https://github.com/symfony/twig-bridge
> Could you try and tell us if it's OK ?

I don't think that's the issue. The handler that displays this error message is part of the TwigBridge.

**Comment by wnoveno** [comment]
Narrowed down the issue. 
I was trying to create dynamic workflows using the DefinitionBuilder.

```
$definitionBuilder = new DefinitionBuilder();
$definition = $definitionBuilder->addPlaces(['draft', 'review', 'rejected', 'published'])
    // Transitions are defined with a unique name, an origin place and a destination place
    ->addTransition(new Transition('to_review', 'draft', 'review'))
    ->addTransition(new Transition('publish', 'review', 'published'))
    ->addTransition(new Transition('reject', 'review', 'rejected'))
    ->build()
;

$marking = new SingleStateMarkingStore('currentState');
$workflow = new Workflow($definition, $marking);
```
But when I used the static approach via the config.yml. The error disappeared. 
So the issue is It's not possible to use the DefinitionBuilder along side the twig functions

**Comment by Simperfit** [comment]
Like @xabbuh said, could you please create a little project with only the workflow code, the controller and the twig template, so we can investigate ?

I've created a reproducer based on what you said, but it seems that there are no bug.

You need to register the workflow in the Registry. Please see https://github.com/Simperfit/symfony-reproducer/commit/ba9280a19c369e2bcccd1d11dd110f2ea35d9972 for 4.1 and see https://github.com/Simperfit/symfony-reproducer/commit/14b50be75375ed2839d6d085d727a957de0b6cf9#diff-5203bc97ede43f8b1dd37565b8fd5c3e for 3.4 which does not changes alot and produce the same result which is `to_review`.

**Comment by lyrixx** [comment]
Ok. I understand your issue. When you are using the config.yml; Symfony is able to know if the application has some workflows configured. So if there are, [it loads the twig extension](https://github.com/lyrixx/symfony/blob/89d1b65037172bfb8bc4de1eeb438cd5c8de80da/src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/ExtensionPass.php#L112-L116) configuration **and** [it adds each workflow](https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L516-L525) in the WorkflowRegistry.

So basically as you are doing everything by hand, you will have to register the twig extension yourself, then to add your workflow in the WorkflowRegistry.

Bonus: The same mechanism applies to the commands related to the workflow.

I'm going to close this issue, but don't hesitate to ask question if you miss something.

**Comment by wnoveno** [comment]
Understood and got it working by registering the twig extension manually. 
An aside though, should the documentation reflect this use case?

**Comment by lyrixx** [comment]
@wnoveno Sure. Would you mind submit a pull request ?
